packages/tideline/js/plot/basal.js

```diff
@@ -169,19 +174,23 @@ function plotBasal(pool, opts = defaults) {
 
     selection.append("rect")
       .attr({
+        id: (d) => `basal_element_${invisible ? "invisible" : "visible"}_${d.id}`,
         x: basal.xPosition,
         y: yPosFn,
-        opacity: function(d) {
+        opacity: (d) => {
+          if (invisible) {
+            return null;
+          }
           if (d.deliveryType === "temp" && d.rate > 0) {
             return opts.opacity - opts.opacityDelta;
           }
           return opts.opacity;
         },
         width: basal.width,
         height: heightFn,
-        class: function(d) {
+        class: (d) => {
           return invisible ? "d3-basal d3-basal-invisible" : `d3-basal d3-rect-basal d3-basal-${d.deliveryType}`;
-        }
+        },
```

Comment:
It can be shortened 🙂
```suggestion
       class: d =>  invisible ? "d3-basal d3-basal-invisible" : `d3-basal d3-rect-basal d3-basal-${d.deliveryType}`,
```